src/components/Select/components/Control/Control.tsx

```diff
@@ -36,18 +58,54 @@ export const Control = (props: ControlProps) => {
         controlInlineStyles.width = width;
     }
 
+    const getControlNode = React.useCallback(() => {
+        return (ref as React.RefObject<HTMLButtonElement>)?.current;
+    }, [ref]);
+
+    const handleClick = () => {
+        dispatch({type: 'SET_ACTIVE', payload: {active: !active}});
+    };
+
+    const handleKeyDown = (e: React.KeyboardEvent<HTMLButtonElement>) => {
+        listboxRef?.current?.onKeyDown(e);
+    };
+
+    const handleResize = React.useCallback(() => {
+        const controlRect = getControlNode()?.getBoundingClientRect();
```

Comment:
`ref` can be used directly here without accessor function, `ref` is always the same object

Reply:
Yes, the same. It\`s something like shortcut to deal with type cast `ref as React.RefObject<HTMLButtonElement>` case of typescript waits here `ref` from `React.forwardRef`